docs/index.md

```diff
@@ -1,54 +1,170 @@
----
-jupytext:
-  cell_metadata_filter: -all
-  formats: md:myst
-  text_representation:
-    extension: .md
-    format_name: myst
-    format_version: 0.12
-    jupytext_version: 1.7.1
-kernelspec:
-  display_name: Python 3
-  language: python
-  name: python3
----
-
-# magicgui 🧙
-
-[![License](https://img.shields.io/pypi/l/magicgui.svg)](https://github.com/pyapp-kit/magicgui/blob/main/LICENSE)
-[![Version](https://img.shields.io/pypi/v/magicgui.svg)](https://pypi.python.org/pypi/magicgui)
-[![conda-forge](https://img.shields.io/conda/vn/conda-forge/magicgui)](https://anaconda.org/conda-forge/magicgui)
-[![Python Version](https://img.shields.io/pypi/pyversions/magicgui.svg)](https://python.org)
-
-Type-based GUI autogeneration for python.
-
-## installation
-
-```shell
-# via pip
-pip install magicgui
-
-# via conda
-conda install -c conda-forge magicgui
-```
+# Overview
+
+`magicgui` is a python library for building graphical user interfaces (GUIs).
+
+It aims to speed up data workflows by providing a simple, consistent API for
+creating GUIs to control various data types.  Graphical user interfaces can
+be built manually using a variety of widgets, or generated automatically from
+python type hints.
+
+## magicgui organization
+
+**magicgui** can be thought of as performing two distinct tasks:
+
+1. Providing an simplified abstraction layer for GUI frameworks, allowing you to
+   use the same API to create UIs that can move between environments and
+   frameworks (such as a desktop app, or a Jupyter notebook).
+2. Providing a mapping of python types to widgets, allowing you to autogenerate
+   graphical user interfaces for functions and dataclasses by annotating with
+   standard python [type hints](https://peps.python.org/pep-0484/).
+
+While there is some degree of customizeability, the emphasis is on rapid
+development of relatively simple GUIs, with minimal boilerplate.  For highly
+customized GUIs, it may be more appropriate to use a lower-level GUI framework.
+
+The API is organized into multiple levels:
+
+<figure markdown>
+  ![Image title](images/api_levels.png){ width="350",loading=lazy }
+  <figcaption>magicgui API layers</figcaption>
+</figure>
+
+### Python Type Mapping ## {: .rectHeading .yellowRect }
+
+At a higher level, **magicgui** provides a mapping of Python types to widgets.
+This API allows you to create graphical user interfaces for your functions and
+dataclasses simply by annotating them with standard python
+[type hints](https://peps.python.org/pep-0484/).
 
-## from types to widgets
+??? info "Type hints??"
```

Comment:
Is this bit formatting for the type hints section? Or do the ??? question marks indicate that this section needs to be returned to and more info added?

I looked in the [mkdocs docs pages](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown) but didn't find anything that looked like this. There were some formatting examples [in the plaintext](https://raw.githubusercontent.com/mkdocs/mkdocs/master/docs/user-guide/writing-your-docs.md) but those were different, like:
```
WARNING:
A warning text block...
```
or,
```
NOTE:
Something for users to note...
```

Reply:
thanks @GenevieveBuckley 

the `???` syntax is a [collapsible block](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks)  it comes from [pymdown-extensions](https://facelessuser.github.io/pymdown-extensions/extensions/details/)

it renders like this:

<img width="713" alt="Screen Shot 2023-02-12 at 9 04 22 PM" src="https://user-images.githubusercontent.com/1609449/218354707-d204932f-cdfd-4057-9f8f-d457b58e9a75.png">

and when clicked:

![Uploading Screen Shot 2023-02-12 at 9.04.32 PM.png…]()